### PR TITLE
Use SI units in parameters when possible (part one, no UTF8 characters)

### PR DESCRIPTION
**Pull request by amilcarlucas**
This first part of the patch only contains ansi characters and should therefore be easier to digest :)

### CODE REVIEW

**Comment by WickedShell** [comment]
Still not a fan of shortening the units, I think for GCS's that aren't
doing unit conversion this makes it less obvious when showing the user.

**Comment by amilcarlucas** [comment]
The patch also:
- adds missing units
- fix wrong units

But again the main issue here is consistency

**Comment by OXINARF** [comment]
> But again the main issue here is consistency

Then why did you insist in shorting the units? I mean, three people said they didn't like the unit shorting, one person said maybe UTF8 characters weren't a good idea (I also though of it, but it compiled and most recent editors will understand UTF8 so I didn't care) and you've only cared about the last one...

It's great it adds missing units and fixes wrong ones, but the main issue that people complained wasn't addressed. Let me also say that I didn't see many inconsistencies (I guess you are considering *Meters* and *meters* to be inconsistent?).

**Comment by lvale** [comment]
We've discussed a bit on this yesterday and I really believe that on the display fields the shortening of the name of the unit to the SI notation while scientifically correct will just add more confusion to users, so I would vote to just keep the descriptions consistent (ie keeping seconds, meters, percentage, etc instead of s, m, %)

**Comment by DonLakeFlyer** [comment]
Just keep in mind that these units gets displayed to the user in many places (at least im QGC they do). For example:
screen shot 2017-01-07 at 9 56 18 am

All the units you see on this screen come from ArduPilot parameter metadata. So you have to be careful they don't get too long. You sort of see it with "Centimeters" on this screen. Is that really any clearer than "cm"? In some cases bigger words adds to more overall noise on a user interface.

Contrived example: you see "cm/s" on this screen. Changing it to "centimeters per second" is going to cause problems and take up a lot of ui space. This comes into play especially if you are running on small screens like a phone.

So it's a tradeoff. More descriptive is not always better.

**Comment by rmackay9** [comment]
I've merged in some of the changes.  There's a few things I didn't merge in:

- didn't merge shortening of "meters" to "m", "centimeters" to "cm", etc.
- didn't merge in some changes to the AP_NavEKF2/3 libraries that I think maybe should go in.  I will ask @priseborough to confirm if these should be changed or not.

**Comment by priseborough** [comment]
I'm happy with the EKF changes that have gone in.

**Comment by amilcarlucas** [comment]
Hi Paul,

There are more EKF on a new pull request from today.

Mit freundlichen Grüßen,

Amilcar Lucas

Software and Algorithms, MD-M1
Business Area Powertrain Mechatronics Diesel

IAV GmbH 
Nordhoffstraße 5
38518 GIFHORN
GERMANY

Tel.: +49 5371 805-3393
Fax: +49 5371 805-1789

E-mail: <mailto:amilcar.lucas@iav.de>
Internet: http://www.iav.com

Sitz/Registered Office: Berlin, 
Registergericht/Registration Court: Amtsgericht Charlottenburg, 
Registernummer/Company Registration Number: HRB 21 280, 
Geschäftsführer/Managing Directors: Kurt Blumenröder, Michael Schubert, 
Kai-Stefan Linnenkohl
Vorsitzender des Aufsichtsrates/Chairman of the Supervisory Board: Prof. 
Dr. Harald Ludanek



Von:    Paul Riseborough <notifications@github.com>
An:     ArduPilot/ardupilot <ardupilot@noreply.github.com>, 
Kopie:  amilcarlucas <amilcar.lucas@iav.de>, Author 
<author@noreply.github.com>
Datum:  09.01.2017 14:00
Betreff:        Re: [ArduPilot/ardupilot] Use SI units in parameters when 
possible (part one, no UTF8 characters) (#5517)



I'm happy with the EKF changes that have gone in.
?

File: libraries/AP_NavEKF2/AP_NavEKF2.cpp [hunk 1]
```diff
@@ -439,7 +439,7 @@ const AP_Param::GroupInfo NavEKF2::var_info[] = {
     // @Range: 0.05 1.0
     // @Increment: 0.05
     // @User: Advanced
-    // @Units: gauss
```
Comment (amilcarlucas) [inline libraries/AP_NavEKF2/AP_NavEKF2.cpp]: Is this correct ?? take a look at it

Comment (priseborough) [inline libraries/AP_NavEKF2/AP_NavEKF2.cpp]: yaw innovations have units of radians

**Comment by amilcarlucas** [comment]
Please re-open this one, the PR branch has been updated

File: libraries/AP_NavEKF2/AP_NavEKF2.cpp (hunk 1 above)
Comment (amilcarlucas) [inline libraries/AP_NavEKF2/AP_NavEKF2.cpp]: This has AFAIK been corrected on the new PR

**Comment by amilcarlucas** [comment]
Could not reopen this, so I made a new one #6182

**Pull request closed without merging**